### PR TITLE
Bug fix for ConvertTo-ARMTemplate

### PR DESCRIPTION
**Pull request by MariusStorhaug**
# Change

- Only clean up `deploy.json` when bicep exists before the conversion.
- Don't attempt to replace workflow or pipeline files if they do not exist. (i.e folders for ado or gh is removed)

## Type of Change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update (Wiki)

## Checklist

- [x] I'm sure there are no other open Pull Requests for the same update/change
- [x] My corresponding pipelines / checks run clean and green without any errors or warnings
- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (readme)
- [x] I did format my code

### CODE REVIEW

File: utilities/tools/ConvertTo-ARMTemplate.ps1 [hunk 1]
```diff
@@ -66,13 +66,15 @@ if ($ConvertChildren) {
...
-Write-Verbose "Remove existing deploy.json files - Remove [$($JsonFilesToRemove.count)] file(s)"
-if ($PSCmdlet.ShouldProcess("[$($JsonFilesToRemove.count)] deploy.json files(s) in path [$armFolderPath]", 'Remove-Item')) {
-    $JsonFilesToRemove | Remove-Item -Force
+if (Test-Path -Path (Join-Path -Path $armFolderPath -ChildPath 'deploy.bicep')) {
+    $JsonFilesToRemove = Get-ChildItem -Path $armFolderPath -Filter 'deploy.json' -Recurse -Force -File
+    Write-Verbose "Remove existing deploy.json files - Remove [$($JsonFilesToRemove.count)] file(s)"
+    if ($PSCmdlet.ShouldProcess("[$($JsonFilesToRemove.count)] deploy.json files(s) in path [$armFolderPath]", 'Remove-Item')) {
+        $JsonFilesToRemove | Remove-Item -Force
+    }
+    Write-Verbose 'Remove existing deploy.json files - Done'
```
Comment (MariusStorhaug) [inline utilities/tools/ConvertTo-ARMTemplate.ps1:75]: Added Test-Path to only delete json files if the bicep file exist, ensuring that the conversion runs clean. Also allowing customers to bring in new modules, running the script again and only having it convert the new module without 'destroying' the entire repo (removing all deploy.json files)

File: utilities/tools/ConvertTo-ARMTemplate.ps1 [hunk 2]
```diff
@@ -173,29 +175,33 @@ if (-not $SkipPipelineUpdate) {
...
+        $ghWorkflowFilesToUpdate = Get-ChildItem -Path $ghWorkflowFolderPath -Filter 'ms.*.yml' -File -Force
+        Write-Verbose ('Update workflow files - Processing [{0}] file(s)' -f $ghWorkflowFilesToUpdate.count)
+        if ($PSCmdlet.ShouldProcess(('[{0}] ms.*.yml file(s) in path [{1}]' -f $ghWorkflowFilesToUpdate.Count, $ghWorkflowFolderPath), 'Set-Content')) {
+            # parallelism is not supported on GitHub runners
+            #$ghWorkflowFilesToUpdate | ForEach-Object -ThrottleLimit $env:NUMBER_OF_PROCESSORS -Parallel {
+            $ghWorkflowFilesToUpdate | ForEach-Object {
+                $content = $_ | Get-Content
+                $content = $content -replace 'templateFilePath:(.*).bicep', 'templateFilePath:$1.json'
+                $_ | Set-Content -Value $content
+            }
```
Comment (MariusStorhaug) [inline utilities/tools/ConvertTo-ARMTemplate.ps1:188]: Added test to check if workflow folder exists before doing cleanup of files in the folder.

File: utilities/tools/ConvertTo-ARMTemplate.ps1 [hunk 3]
```diff
@@ -173,29 +175,33 @@ if (-not $SkipPipelineUpdate) {
...
+        $adoPipelineFilesToUpdate = Get-ChildItem -Path $adoPipelineFolderPath -Filter 'ms.*.yml' -File -Force
+        Write-Verbose ('Update Azure DevOps pipeline files - Processing [{0}] file(s)' -f $adoPipelineFilesToUpdate.count)
+        if ($PSCmdlet.ShouldProcess(('[{0}] ms.*.yml file(s) in path [{1}]' -f $adoPipelineFilesToUpdate.Count, $adoPipelineFolderPath), 'Set-Content')) {
+            # parallelism is not supported on GitHub runners
+            #$adoPipelineFilesToUpdate | ForEach-Object -ThrottleLimit $env:NUMBER_OF_PROCESSORS -Parallel {
+            $adoPipelineFilesToUpdate | ForEach-Object {
+                $content = $_ | Get-Content
+                $content = $content -replace 'templateFilePath:(.*).bicep', 'templateFilePath:$1.json'
+                $_ | Set-Content -Value $content
+            }
```
Comment (MariusStorhaug) [inline utilities/tools/ConvertTo-ARMTemplate.ps1:204]: Added test to check if `.azuredevops\modulePipelines` folder exists before doing cleanup of files in the folder.

**Review by eriqua — APPROVED** [review]
🆗

**Pull request merged**